jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/FeatureSizeValidatingSuite.scala

```diff
@@ -65,8 +65,6 @@ class FeatureSizeValidatingSuite extends FunSuite with PerTest {
         (id, lp.label, lp.features)
     }.toDF("id", "label", "features")
     val xgb = new XGBoostClassifier(paramMap)
-    intercept[Exception] {
```

Comment:
@trivialfis, Could you enable some checking in native XGBoost?

Reply:
Could you please be more specific?

Reply:
With this PR, xgboost will not throw exception even the feature size is different in each xgboost task. I recall xgboost native has disabled the checking for feature size.

Reply:
Internally XGBoost obtains the maximum feature size across all workers.

Reply:
so the question is, if the DMatrix building is out of Rabit context, how xgboost obtains the maximum feature size?

Reply:
It's obtained right before the training starts but it's not really reliable.

Reply:
It's my fault. When moving the dmatrix building into Rabit, the `rabit::Allreduce<rabit::op::Max>(&info_.num_col_, 1);` will work, and the num_col_ will be the correct value. So no exception thrown when checking the feature size.